### PR TITLE
[SATURN-1260] Adding Project-Specific view to cluster config drawer

### PR DESCRIPTION
**Pull request by ehanna4**
DO NOT MERGE UNTIL APPROVED BY PO

This ticket holds all the renaming of "Notebook Runtime" to "Application Compute" and will expose the RStudio Project-specific feature to users. Need to coordinate with PO before merging. Waiting on link to zendesk page to be made public.

Tested by:
- creating, deleting and replacing a variety of clusters from the Jupyter, Custom and Project-specific environments. 
- observed that you are able to paste the same image urls in either Custom or Project-specific environment (and validation works the same) but when you reopen the drawer it will open to the environment you have a current cluster on.
- observed the rstudio logo appears in the widget when using an appropriate image

### CODE REVIEW

File: src/components/NewClusterModal.js
Comment (panentheos) [inline src/components/NewClusterModal.js]: TOL: Do we want a separate piece of state for the project specific image?

It's functionally identical to the custom env image state, in terms of how it's ultimately used. Having it separate means that you can switch between custom and project-specific, and retain a separate temporary value for each. If that isn't a particularly important part of the UX, we could just use the same state value for both, which would eliminate some of the conditional logic around which value to use.

File: src/components/NewClusterModal.js
Comment (panentheos) [inline src/components/NewClusterModal.js]: This doesn't have to be an RStudio image, right? Should this just have the same message as the other one?

File: src/components/NewClusterModal.js
Comment (panentheos) [inline src/components/NewClusterModal.js]: nit: this code spacing is confusing when coupled with this particular text. I think expanding the children array fully would help, e.g.:
```
div([
  X,
  X,
  X
])
```

File: src/components/NewClusterModal.js
Comment (panentheos) [inline src/components/NewClusterModal.js]: nit: this `Fragment` is unnecessary since it only contains a single child.

File: src/components/NewClusterModal.js
Comment (ehanna4) [reply src/components/NewClusterModal.js]: I struggled with this decision and went back and forth about just using the customEnvImage. I ultimately went with two pieces of state so that the custom and project-specific environments would be more decoupled as it seemed to be emphasized in the requirement that this BYORstudio thing should be clearly separated from existing notebook stuff, but I am certainly open to changing it

File: src/components/NewClusterModal.js
Comment (ehanna4) [reply src/components/NewClusterModal.js]: Good catch yes I will fix that

File: src/components/NewClusterModal.js
Comment (ehanna4) [reply src/components/NewClusterModal.js]: Upon further reflection and discussion with Isaac I will remove the extra state and just use customEnvImage. Thanks for the call out!

**Comment by ehanna4** [comment]
UPDATE: will be reverting to "Runtime" again while we wait on final-final approval of title change

File: src/components/common.js
Comment (panentheos) [inline src/components/common.js]: Since we already pass all additional props through to the underlying library, we could remove this special handling and just have the consumer pass `maxMenuHeight`, which would have the same effect.

File: src/pages/workspaces/workspace/applications/AppLauncher.js
Comment (panentheos) [inline src/pages/workspaces/workspace/applications/AppLauncher.js]: nit: grammar

File: src/pages/workspaces/workspace/Notebooks.js
Comment (panentheos) [inline src/pages/workspaces/workspace/Notebooks.js]: Is this change intended?

File: src/components/NewClusterModal.js
Comment (panentheos) [inline src/components/NewClusterModal.js]: Is this change intended?

File: src/components/NewClusterModal.js
Comment (panentheos) [inline src/components/NewClusterModal.js:210]: This means that if you switch to a prebuilt image and then back to custom, it loses any url you entered. Is that what we want?

**Review by panentheos — APPROVED** [review]
Looks good overall, see a few questions.

File: src/components/NewClusterModal.js
Comment (ehanna4) [reply src/components/NewClusterModal.js:210]: This was intended. While the functionality of custom env and pre-built are super similar there should appear to be a difference between them to users. This way the state is reduced but its still made clear there is a difference between these two options.

**Pull request merged**
